runtime/parachains/src/scheduler.rs

```diff
@@ -146,19 +151,10 @@ impl<T: Config> Pallet<T> {
 			},
 		);
 
-		Self::reschedule_occupied_cores(AvailabilityCores::<T>::get());
-		Self::clear_claimqueue();
+		Self::push_occupied_cores_to_assignment_provider(n_cores);
+		Self::push_claimqueue_items_to_assignment_provider();
```

Comment:
> When we push back, it goes to the front of the assignment provider.

That's an implementation detail of the provider itself. Nothing the scheduler knows or should be aware of.

Reply:
How is this an implementation detail? Common sense would be that we push back in the same order as we popped. In any case it needs to be specified, it cannot be an implementation detail.

Reply:
If you look at any data structure, you also have it specified where stuff goes if you push. E.g. push_back, pop_front ... because it matters. It cannot just be an implementation detail of `Vec`.

Reply:
In my view it was an implementation detail of the assignment provider. Super easy to push back in return, of course.